integration_test/integration_test.go

```diff
@@ -298,6 +372,621 @@ func (t *WASMIntegrationTestSuite) Test04_ContractExecution() {
+			stakingtype.NewDescription("zeroreward_validator_3", "", "", "", ""),
+		)
+
+		assert.NoError(t.T(), err)
+
+		err = applyVoteTallyingProposal(
+			desc.GetConnectionWithContext(context.Background()),
+			proposalContent,
+			t.ZeroRewardValidatorWallet3,
+			[]*WalletInfo{t.ValidatorWallet1, t.ValidatorWallet2, t.ValidatorWallet3, t.ValidatorWallet4},
+		)
+
+		assert.NoError(t.T(), err)
+	}
+
+	{
+		fmt.Println("Check existence & status of the zero reward validator")
+
+		client := zrValidatorType.NewQueryClient(desc.GetConnectionWithContext(context.Background()))
+		validatorStatus, err := client.ZeroRewardValidators(context.Background(), &zrValidatorType.QueryZeroRewardValidatorsRequest{})
+		assert.NoError(t.T(), err)
+
+		thisVoluntaryValAddress := sdktypes.ValAddress(t.ZeroRewardValidatorWallet3.ByteAddress).String()
+
+		if assert.Contains(t.T(), validatorStatus.GetZeroRewardValidators(), thisVoluntaryValAddress) {
+			fmt.Println(thisVoluntaryValAddress, "successfully participated in the validator set!")
+		} else {
+			fmt.Println(thisVoluntaryValAddress, "is not found")
+			t.T().Fail()
+		}
+	}
+
+	{
+		fmt.Println("Add normal validator")
+
+		// more than zero reward validator but less than other validator
+		delegationAmt := sdktypes.NewCoin("axpla", sdktypes.NewInt(1_500000_000000_000000))
+
+		createValidatorMsg, err := stakingtype.NewMsgCreateValidator(
+			sdktypes.ValAddress(t.ValidatorWallet5.ByteAddress.Bytes()),
+			&ed25519.PubKey{Key: t.Validator5PVKey.PubKey.Bytes()},
+			delegationAmt,
+			stakingtype.NewDescription("validator5", "", "", "", ""),
+			stakingtype.NewCommissionRates(
+				sdktypes.MustNewDecFromStr("0.1"),
+				sdktypes.MustNewDecFromStr("0.2"),
+				sdktypes.MustNewDecFromStr("0.01"),
+			),
+			sdktypes.NewInt(1),
+		)
+
+		assert.NoError(t.T(), err)
+
+		feeAmt := sdktypes.NewDec(xplaProposalGasLimit).Mul(sdktypes.MustNewDecFromStr(xplaGasPrice))
+		fee := sdktypes.Coin{
+			Denom:  "axpla",
+			Amount: feeAmt.Ceil().RoundInt(),
+		}
+
+		txhash, err := t.ValidatorWallet5.SendTx(ChainID, createValidatorMsg, fee, xplaProposalGasLimit, false)
+		if assert.NotEqual(t.T(), "", txhash) && assert.NoError(t.T(), err) {
+			fmt.Println("Tx sent", txhash)
+		} else {
+			fmt.Println(err)
+		}
+
+		err = txCheck(txhash)
+		if assert.NoError(t.T(), err) {
+			fmt.Println("Tx applied", txhash)
+		} else {
+			fmt.Println(err)
+		}
+	}
+
+	fmt.Println("Waiting 13sec (> 2 block time) for the validator status refresh...")
+	time.Sleep(time.Second * 13)
+
+	{
+		fmt.Println("Check the # of the validator set")
+
+		prevoteCnt, err := getMaxPrevoterCounts()
+		assert.NoError(t.T(), err)
+
+		if assert.Equal(t.T(), maxValidators+1, prevoteCnt) {
+			fmt.Println("Prevote reaches to the right number:", prevoteCnt)
+		} else {
+			fmt.Println("Lack of the # of prevotes. Expect:", maxValidators+1, "Actual:", prevoteCnt)
+			t.T().Fail()
+		}
+	}
+
+	{
+		fmt.Println("Try to turn off the zero reward validator")
```

Comment:
It seems necessary to check whether the "Zero Reward Validator" operates normally in both cases when it belongs to the active set and when it does not belong to it.

Reply:
In addition, it is necessary to check the status of the case where the zero reward validator enters the active set or not in the jail situation of the general validator.

Reply:
Could `voluntary validator` be in inactive set? I thought it should always be in active set...?

Reply:
https://github.com/xpladev/xpla/pull/84/commits/c4ef4ccdcf1603bc2fe5e319e93ef804018c224d